issue_processing/issue_processing.py

```diff
@@ -128,17 +253,107 @@ def reformat(issue_data):
+                    resolution_event["user"] = issue["user"]
+                    resolution_event["created_at"] = event["created_at"]
+                    resolution_event["event"] = "resolution_updated"
+                    resolution_event["event_info_1"] = label
+                    resolution_event["event_info_2"] = ""
+                    resolution_event["ref_target"] = ""
+                    issue["eventsList"].append(resolution_event)
+
+                    # the added resolution is appended to the old resolution list
+                    # works because label events are sorted by time
+                    new_resolutions = list(resolution_changes[-1][1])
+                    new_resolutions.append(str(label))
+                    # creates new resolution change object
+                    resolution_changes.append([resolution_event["created_at"], new_resolutions])
+
+            # %TODO: event for removed labels has to be checked. Data is missing
+
+            # if event_info_1 does not exist in the event, an empty info is created
+            if "event_info_1" not in event:
+                event["event_info_1"] = ""
+
+            # if event_info_2 does not exist in the event, an empty info is created
+            if 'event_info_2' not in event:
+                event["event_info_2"] = ""
+
+        # state and resolution change lists get sorted by time
+        state_changes.sort(key=lambda x: x[0])
+        resolution_changes.sort(key=lambda x: x[0])
+
+        # the format of every comment is adjusted to the event format
+        for comment in issue["commentsList"]:
+            comment["event"] = "commented"
+            comment["ref_target"] = ""
+
+            comment["created_at"] = format_time(comment["created_at"])
+
+            # cache comment by date to resolve/re-arrange references later
+            comments[comment["created_at"]] = comment
+
+            # the state the issue had when the comment was written is searched out
+            for state in state_changes:
+                if comment['created_at'] > state[0]:
+                    comment['event_info_1'] = state[1]
+
+            # the resolution the issue had when the comment was written is searched out
+            for resolution in resolution_changes:
+                if comment['created_at'] > resolution[0]:
+                    comment['event_info_2'] = resolution[1]
+
         # merge events and comment lists
         issue["eventsList"] = issue["commentsList"] + issue["eventsList"]
 
-        # add 'closed_at' information if not present yet
-        if issue["closed_at"] is None:
-            issue["closed_at"] = ""
-
         # remove events without user
         issue["eventsList"] = [event for event in issue["eventsList"] if
                                not (event["user"] is None or event["ref_target"] is None)]
 
+        # sorts eventsList by time
+        issue["eventsList"].sort(key=operator.itemgetter('created_at'))
```

Comment:
Before I look at the details of this PR myself, one small comment and advice here: Python has a style guide ([PEP8](https://www.python.org/dev/peps/pep-0008/); unlike R), which is also directly supported by PyCharm. You just need to re-format the file (press <kbd>Alt+Shift+L</kbd>).
I highly recommend using PyCharm's formatter. This will solve most of the formatting issues up-front (except for quotation marks, I suppose).

Why I comment here: The style guide states the following – in contrast to our coding conventions from the network library:
> Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.

Reply:
Thank you for this hint @clhunsen.